src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxIterationCountTests.CustomAppDomainDataLimit.cs

```diff
@@ -0,0 +1,140 @@
+        {
+            _ = iterationCount;
+            _ = blob;
+
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            RemoteExecutor.Invoke((certName) =>
+            {
+                AppDomain.CurrentDomain.SetData("System.Security.Cryptography.Pkcs12UnspecifiedPasswordIterationLimit", 0);
+
+                PfxInfo pfxInfo = s_certificatesDictionary[certName];
+
+                CryptographicException ce = Assert.Throws<CryptographicException>(() => Import(pfxInfo.Blob));
+                Assert.Contains("2233907", ce.Message);
+            }, name).Dispose();
+        }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(memberName: nameof(PfxIterationCountTests.GetCertsWith_IterationCountExceedingDefaultLimit_MemberData), MemberType = typeof(PfxIterationCountTests))]
+        public void Import_AppDomainDataWithValueMinusOne_IterationCountExceedingDefaultLimit(string name, string password, bool usesPbes2, byte[] blob, long iterationCount)
+        {
+            _ = password;
+            _ = blob;
+            _ = iterationCount;
+
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            RemoteExecutor.Invoke((certName) =>
+            {
+                AppDomain.CurrentDomain.SetData("System.Security.Cryptography.Pkcs12UnspecifiedPasswordIterationLimit", -1);
+
+                PfxInfo pfxInfo = s_certificatesDictionary[certName];
+
+                if (OperatingSystem.IsWindows())
+                {
+                    // Opting-out with AppDomain data value -1 will still give us error because cert is beyond Windows limit.
+                    // But we will get the CryptoThrowHelper+WindowsCryptographicException.
+                    PfxIterationCountTests.VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(pfxInfo.Blob));
+                }
+                else
+                {
+                    Assert.NotNull(Import(pfxInfo.Blob));
+                }
+            }, name).Dispose();
+        }
+
+        public static readonly Dictionary<string, PfxInfo> s_certificatesDictionary
+            = PfxIterationCountTests.s_Certificates.ToDictionary((c) => c.Name);
+    }
+
+    public class PfxIterationCountTests_CustomLimit_X509Certificate2 : PfxIterationCountTests_CustomAppDomainDataLimit
+    {
+        internal override X509Certificate Import(byte[] blob) => new X509Certificate2(blob);
+    }
+
+    public class PfxIterationCountTests_CustomLimit_X509Certificate2Collection : PfxIterationCountTests_CustomAppDomainDataLimit
```

Comment:
The custom limit tests only test X509Certificate2Collection and X509Certificate2.  The non-custom-limit tests also test `X509Certificate` (no 2).  Should these also?

Reply:
We do test X509Certificate (no 2), it is performed on the base `PfxIterationCountTests_CustomAppDomainDataLimit`, it is just that base can't be abstract because of RemoteExecutor needs to instantiate the class where it is being used, so base is just a concrete type with a virtual, hence the comment:
https://github.com/dotnet/runtime/blob/008f501c8714c37ea2ddfcb63037ec72ecfe1cf4/src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxIterationCountTests.CustomAppDomainDataLimit.cs#L17-L18